match/lib/match/runner.rb

```diff
@@ -222,20 +244,23 @@ def fetch_provisioning_profile(params: nil, certificate_id: nil, app_identifier:
       end
 
       # Install the provisioning profiles
-      profile = profiles.last
+      stored_profile_path = profiles.last
       force = params[:force]
 
+      portal_profile = nil
+      portal_profile = self.cache.portal_profile(stored_profile_path: stored_profile_path, keychain_path: keychain_path) if stored_profile_path
+
       if params[:force_for_new_devices]
-        force = should_force_include_all_devices(params: params, prov_type: prov_type, profile: profile, keychain_path: keychain_path) unless force
+        force = ProfileIncludes.should_force_include_all_devices?(params: params, portal_profile: portal_profile, cached_devices: self.cache.devices) unless force
       end
 
       if params[:include_all_certificates]
         # Clearing specified certificate id which will prevent a profile being created with only one certificate
         certificate_id = nil
-        force = should_force_include_all_certificates(params: params, prov_type: prov_type, profile: profile, keychain_path: keychain_path) unless force
+        force = ProfileIncludes.should_force_include_all_certificates?(params: params, portal_profile: portal_profile, cached_certificates: self.cache.certificates) unless force
```

Comment:
we could also rewrite this as `force ||=`

Reply:
Done in a62e7fc.